### PR TITLE
 added tests for shopping cart functionality

### PR DESCRIPTION
**Pull request by annton-shakur**

### CODE REVIEW

File: src/test/java/mate/academy/onlinebookstoreproject/repository/ShoppingCartRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        shoppingCart.setCartItems(Set.of(cartItem));
+    }
+
+    @Test
+    @DisplayName("Return cart by a valid user id")
+    void findByUserId_WithValidId_ReturnShoppingCart() {
+        ShoppingCart expected = shoppingCart;
+        ShoppingCart actual = shoppingCartRepository.findByUserId(VALID_ID).get();
+        assertEquals(expected, actual);
+        assertEquals(Set.of(cartItem), actual.getCartItems());
```
Comment (yevheniisynytsia) [inline src/test/java/mate/academy/onlinebookstoreproject/repository/ShoppingCartRepositoryTest.java:98]: TBH comparing shopping carts is enough, but you can leave as it is also

File: src/test/java/mate/academy/onlinebookstoreproject/service/impl/ShoppingCartServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    private static ShoppingCartRepository shoppingCartRepository;
+    @InjectMocks
+    private static ShoppingCartServiceImpl shoppingCartService;
+    private static Book theGreatGatsbyBook;
+    private static Category categoryOne;
+    private static ShoppingCart shoppingCart;
+    private static ShoppingCartResponseDto shoppingCartResponseDto;
+    private static CartItem cartItem;
+    private static CartItemResponseDto cartItemResponseDto;
+    private static User user;
```
Comment (yevheniisynytsia) [inline src/test/java/mate/academy/onlinebookstoreproject/service/impl/ShoppingCartServiceImplTest.java:34]: place those before mocks

File: src/test/java/mate/academy/onlinebookstoreproject/service/impl/ShoppingCartServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        shoppingCartResponseDto.setId(1L);
+        shoppingCartResponseDto.setCartItems(Set.of(cartItemResponseDto));
+    }
+
+    @Test
+    @DisplayName("Return shopping cart by user id")
+    void getCartByUsedId_WithValidId_ReturnShoppingCartDto() {
+        Mockito.when(shoppingCartRepository.findByUserId(1L))
+                .thenReturn(Optional.of(shoppingCart));
+        Mockito.when(shoppingCartMapper.toDto(shoppingCart))
```
Comment (yevheniisynytsia) [inline src/test/java/mate/academy/onlinebookstoreproject/service/impl/ShoppingCartServiceImplTest.java:90]: please add static import

File: src/test/resources/database/cart-items/delete-cart-items.sql [hunk 4]
```diff
@@ -0,0 +1 @@
+DELETE FROM cart_items
```
Comment (yevheniisynytsia) [inline src/test/resources/database/cart-items/delete-cart-items.sql]: add empty lines

**Review by yevheniisynytsia — CHANGES_REQUESTED** [review]
Well done! Only small changes requested

File: src/test/java/mate/academy/onlinebookstoreproject/repository/ShoppingCartRepositoryTest.java (hunk 1 above)
Comment (annton-shakur) [reply src/test/java/mate/academy/onlinebookstoreproject/repository/ShoppingCartRepositoryTest.java:98]: i added this for testing whether cart items are fetched additionally

**Review by yevheniisynytsia — APPROVED** [review]

**Pull request merged**